src/v/kafka/server/handlers/fetch.cc

```diff
@@ -106,8 +125,22 @@ static ss::future<read_result> read_from_partition(
     std::unique_ptr<iobuf> data;
     std::vector<cluster::rm_stm::tx_range> aborted_transactions;
     try {
-        auto result = co_await rdr.reader.consume(
-          kafka_batch_serializer(), deadline ? *deadline : model::no_timeout);
+        const auto ts_deadline = deadline.value_or(model::no_timeout);
+        const auto failsafe_deadline = failsafe_deadline_from(ts_deadline);
+        auto consume_fut = rdr.reader.consume(
+          kafka_batch_serializer(), ts_deadline);
+        auto result
+          = co_await ss::with_timeout(failsafe_deadline, std::move(consume_fut))
+              .handle_exception_type(
+                [&reader_config](const ss::timed_out_error& err) {
+                    vlog(
+                      klog.error,
+                      "Tiered-storage reader timed out. Reader config: {}",
```

Comment:
This is positioned as a timeout for tiered storage readers, but isn't this the generic fetch path that would apply to any type of reader?

Reply:
Yes, fixed the message